cmd/bench.go

```diff
@@ -358,7 +375,7 @@ func benchE2E(ctx context.Context, args []string, params benchmarkCommandParams,
 		}
 	}
 
-	url := fmt.Sprintf("http://%s:%s/v1/%v", host, port, path)
+	url := fmt.Sprintf("http://%s:%s/v1/%v", host, strconv.Itoa(port), path)
 	if params.metrics {
```

Comment:
My favourite approach to this is using port 0. The OS will resolve it into a free, unprivileged port. The only trouble is that we'll have to retrieved it from the listener, which could mean some slight refactoring is needed.

Reply:
I thought of that yesterday, and shied away initially because it looked like it'd be painful to recover the port from the runtime's internal address list.

This morning, I tried the "bind port 0" approach out, and learned that the server does not actually bind a socket until a few hundred microseconds *after* it's supposedly initialized. I wound up adding a busy wait loop to recover the real socket address so that the client further down in the test can get the correct port number.